### PR TITLE
feat: Area viz migration

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY
1. Migrate Area Chart to v2
2. fix migration double heads

image

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

```shell
# upgrade
superset db upgrade

# downgrade
superset db downgrade
```

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [x] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/utils/migrate_viz.py [hunk 1]
```diff
@@ -117,6 +111,38 @@ def _pre_action(self) -> None:
...
+            self.data["contributionMode"] = "row"
+
+        stacked = self.data.get("stacked_style")
+        stacked_map = {
+            "expand": "Expand",
+            "stack": "Stack",
+        }
+        if stacked:
+            self.data["show_extra_controls"] = True
+            self.data["stack"] = stacked_map.get(stacked)
```
[the remark this answers is not in the document]
Comment (zhaoyongjie) [reply superset/utils/migrate_viz.py]: done.

File: superset/migrations/versions/2022-07-07_14-00_06e1e70058c7_migrating_legacy_area.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            print(f"Upgrading ({idx}/{total}): {slc.slice_name}#{slc.id}")
+            new_viz = area_processor.upgrade(slc)
+            session.merge(new_viz)
+        except Exception as exc:
+            print(
+                "Error while processing migration: '{}'\nError: {}\n".format(
+                    slc.slice_name, str(exc)
+                )
+            )
+    session.commit()
```
[the remark this answers is not in the document]
Comment (zhaoyongjie) [reply superset/migrations/versions/2022-07-07_14-00_06e1e70058c7_migrating_legacy_area.py:72]: same as the TreeMap migration, optimize write speed.

**Review by michael-s-molina — APPROVED** [review]

**Comment by john-bodley** [comment]
@zhaoyongjie would you mind explaining the purpose of this migration if the "Area Chart" chart type still exists? Is the intention to fully deprecate said chart type? If so this migration would need to be re-run.

**Comment by zhaoyongjie** [comment]
@john-bodley Sorry, I'm stuck in another [PR](https://github.com/apache/superset/pull/20639) with a cypress problem. I think the migration process should be re-run after resolve the frontend migration issue.

**Pull request merged**